flipperdevices/flipperzero-firmware
# Snake Plugin: Store game state on close and restore it on restart, show highscore

# What's new

- Implements #1041 as it safes the current game state and restores it on restart
- Saves also an highscore to an file and shows the highscore on GameOver

# Verification 

- Open App and close it after moving a little bit around, reopen it and the state should be as before. 
- Close the App when GameOver Screen is shown. The Game should restart normal.
- Crash the snake and see if the highscore is shown. Do it again with an higher score and see if it increased, further do it with a lower score to verify if it stays the same.

# Checklist (For Reviewer)
